libr/anal/anal.c

```diff
@@ -118,17 +118,15 @@ R_API RAnal *r_anal_new() {
 	anal->plugins = r_list_newf ((RListFree) r_anal_plugin_free);
 	if (anal->plugins) {
 		for (i=0; anal_static_plugins[i]; i++) {
-			static_plugin = R_NEW (RAnalPlugin);
-			*static_plugin = *anal_static_plugins[i];
-			r_anal_add (anal, static_plugin);
+			r_anal_add (anal, anal_static_plugins[i]);
 		}
 	}
 	return anal;
 }
 
 R_API void r_anal_plugin_free (RAnalPlugin *p) {
 	if (p && p->fini) {
-		p->fini (p);
+		p->fini (NULL);
 	}
```

Comment:
None of the plugins apart from `r_anal_plugin_tms320` have `fini` in the declaration, or they are updated during execution.

And in `r_anal_plugin_tms320` `fini` doesn't use passed pointer for finalizing.